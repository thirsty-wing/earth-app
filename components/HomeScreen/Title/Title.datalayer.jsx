import { useState } from "react";
import Title from "./Title";
import { yieldRotateGlobe } from "./Title.utils";


function TitleDatalayer() {

  const [globeOrientation, setGlobeOrientation] = useState(1);

  const onPress = yieldRotateGlobe(globeOrientation, setGlobeOrientation);

  return(
    <Title
      globeOrientation={globeOrientation}
      onPress={onPress}
    />
  );
}

export default TitleDatalayer;
