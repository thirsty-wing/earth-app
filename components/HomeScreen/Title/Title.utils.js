/**
 * @callback setGlobeOrientation
 * @param {number}
 */

/**
 * yieldRotateGlobe.
 *
 * @param {number} globeOrientation
 * @param {function} setGlobeOrientation
 */
export function yieldRotateGlobe(globeOrientation, setGlobeOrientation) {
  return function() {
    const newGlobeOrientation = globeOrientation > 2
      ? 1
      : globeOrientation + 1;

      setGlobeOrientation(newGlobeOrientation);
  };
}
