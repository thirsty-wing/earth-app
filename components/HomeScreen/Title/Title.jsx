import { Text, Pressable } from "react-native";
import { theme } from "../../../utils/theme";
import { getGlobe } from "./Title.utils";


/**
 * Title.
 *
 * @param {object} props
 * @param {number} props.globeOrientation
 * @param {Function} props.onPress
 */
function Title(props) {

  const {
    globeOrientation,
    onPress,
  } = props;

  const globe = getGlobe(globeOrientation);

  return(
    <Pressable
      style={{
        alignItems: "center",
        width: "100%",
      }}
      onPress={onPress}
    >
      <Text
        style={{
          color: theme.palette.text.primary,
          fontSize: theme.sizes.text.title,
          textAlign: "center",
        }}
      >
        What in the world? {globe}
      </Text>
    </Pressable>
  );
}

export default Title;
