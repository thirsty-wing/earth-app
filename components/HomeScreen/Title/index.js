import Title from "./Title.datalayer";

export default Title;
