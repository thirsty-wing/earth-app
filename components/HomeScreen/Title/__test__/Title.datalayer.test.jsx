import Title from "../Title.datalayer";
import { shallow } from "../../../../utils/test/shallow";


it("should render", () => {
  const content = shallow(<Title/>);

  expect(content).toMatchSnapshot();
});

