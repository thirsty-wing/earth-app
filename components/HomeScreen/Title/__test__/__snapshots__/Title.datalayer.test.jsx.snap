// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Title
  globeOrientation={1}
  onPress={[Function]}
/>
`;
